syntax = "proto3";

option csharp_namespace = "TrackAndTrace.TimeProvider";

service TimeService {
    rpc SetTime(SetTimeRequest) returns (SetTimeResponse);
    rpc ResetTime(ResetTimeRequest) returns (ResetTimeResponse);
    rpc GetCurrentTime(CurrentTimeRequest) returns (CurrentTimeResponse);
}

message SetTimeRequest {
    string time = 1; // ISO 8601 format
}

message SetTimeResponse {
    bool success = 1;
}

message ResetTimeRequest {}

message ResetTimeResponse {
    bool success = 1;
}

message CurrentTimeRequest {}

message CurrentTimeResponse {
    string current_time = 1; // ISO 8601 format
    string previous_time = 2; // ISO 8601 format
}